### PR TITLE
Fix - context timeout not launched on get bucket location #1413

### PR DESCRIPTION
**Pull request by alexcocia**
Using context on request, in order to handle context timeouts

### CODE REVIEW

File: bucket-cache.go [hunk 1]
```diff
@@ -102,6 +102,7 @@ func (c Client) getBucketLocation(ctx context.Context, bucketName string) (strin
 		return "", err
 	}
 
+	req = req.WithContext(ctx)
 	// Initiate the request.
 	resp, err := c.do(req)
 	defer closeResponse(resp)
```
Comment (harshavardhana) [inline bucket-cache.go:108]: it is better to use http.NewRequestWithContext() instead of req.WithContext @alexcocia

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: bucket-cache.go (hunk 1 above)
Comment (harshavardhana) [reply bucket-cache.go:108]: Here is the change I would expect

```diff
git diff
diff --git a/bucket-cache.go b/bucket-cache.go
index 7d485a6..156150f 100644
--- a/bucket-cache.go
+++ b/bucket-cache.go
@@ -97,7 +97,7 @@ func (c Client) getBucketLocation(ctx context.Context, bucketName string) (strin
        }
 
        // Initialize a new request.
-       req, err := c.getBucketLocationRequest(bucketName)
+       req, err := c.getBucketLocationRequest(ctx, bucketName)
        if err != nil {
                return "", err
        }
@@ -169,7 +169,7 @@ func processBucketLocationResponse(resp *http.Response, bucketName string) (buck
 }
 
 // getBucketLocationRequest - Wrapper creates a new getBucketLocation request.
-func (c Client) getBucketLocationRequest(bucketName string) (*http.Request, error) {
+func (c Client) getBucketLocationRequest(ctx context.Context, bucketName string) (*http.Request, error) {
        // Set location query.
        urlValues := make(url.Values)
        urlValues.Set("location", "")
@@ -198,7 +198,7 @@ func (c Client) getBucketLocationRequest(bucketName string) (*http.Request, erro
        }
 
        // Get a new HTTP request for the method.
-       req, err := http.NewRequest(http.MethodGet, urlStr, nil)
+       req, err := http.NewRequestWithContext(ctx, http.MethodGet, urlStr, nil)
        if err != nil {
                return nil, err
        }
```

Comment (alexcocia) [reply bucket-cache.go]: New commit provided applying your code. Thanks a lot, that was a quick response!

File: bucket-cache.go [hunk 2]
```diff
@@ -97,11 +97,12 @@ func (c Client) getBucketLocation(ctx context.Context, bucketName string) (strin
 	}
 
 	// Initialize a new request.
-	req, err := c.getBucketLocationRequest(bucketName)
+	req, err := c.getBucketLocationRequest(ctx, bucketName)
 	if err != nil {
 		return "", err
 	}
 
+	//req = req.WithContext(ctx)
```
Comment (harshavardhana) [inline bucket-cache.go:105]: Not needed since you made the change already can be removed

```suggestion
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by alexcocia** [comment]
Totally right. Forgot to remove it after tests on my env.

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**